### DISCUSSION TITLE
charset_map error

### DISCUSSION THREAD

**Discussion by geryheselmans** [nginx/nginx#193] [Q&A] [answered]
I looking for some advice regarding following error:

2024/09/12 14:55:37 [error] 25#0: *5200198 no "charset_map" between the charsets "utf-8" and "iso-8859-1" while reading response header from upstream, client: xxxx, server: xxxx, request: "GET /xxxx/xxxx/ HTTP/1.1", subrequest: "/xxxx/xxxx", upstream: "https://xx.xx.xx.xx:8080/xxxx/", host: "xxxx"

We see this error randomly popping up and we can’t seem to able to reproduce this. Our nginx is setup as a cache and reverse proxy between 2 internal systems. We know that our 404 errors can be iso-8859-1 encoded, however our 200 response should be utf-8. However we see the error appearing on both status codes and if we redo the same requests later we don’t get the error. We also looked at the headers of the response both 200 and 404 and we don’t see a likely candidate for which header could be the culprit   

What exactly does the error mean? Is it possible to add some debugging to figure out which header is causing this? What is the best way to fix this error?

I redacted the URL, host and ip address for security reasons as these are not public APIs.

**Comment by royteeuwen** [comment]
Any help on this error?

**Comment by Maryna-f5** [comment]
This discussion has been inactive for more than 90 days, closing it.

**Comment by pluknet** [comment, marked as answer]
This may happen when receiving proxied answers that already carry a "utf-8" charset in the “Content-Type” response header,
and the configuration uses a distinct destination charset as set with `charset iso-8859-1;` together with `charset_override`.
In this case, a convertion table is needed from one charset to another charset; the error message is about this.
